modules/rgbd/src/tsdf.cpp

```diff
@@ -434,20 +495,22 @@ void TSDFVolumeCPU::integrate(InputArray _depth, float depthFactor, const cv::Ma
     CV_Assert(_depth.type() == DEPTH_TYPE);
     CV_Assert(!_depth.empty());
     Depth depth = _depth.getMat();
-    IntegrateInvoker ii(*this, depth, intrinsics, cameraPose, depthFactor);
+    Mat pixNorms = preCalculationPixNorm(depth, intrinsics);
```

Comment:
Since `pixNorm` depends on image size only, it shouldn't be updated at each call. We can keep the precalculated `pixNorm` inside a class with intrinsics and frame size and update it only if these parameters are changed.

Reply:
done